### PR TITLE
docs: add mfa methods to the reference docs

### PR DESCRIPTION
**Pull request by kangmingtay**
## What kind of change does this PR introduce?
* Adds the gotrue-js MFA methods to the reference guide

### CODE REVIEW

File: spec/supabase_js_v2.yml [hunk 1]
```diff
@@ -421,6 +421,80 @@ pages:
...
+  auth.mfa.challenge():
+    title: 'mfa.challenge()'
+    $ref: '@supabase/gotrue-js.GoTrueMFAApi.challenge'
+    examples:
+      - name: Creates a challenge for a factor
+        isSpotlight: true
+        js: |
+          ```js
+          const { data, error } = await supabase.auth.mfa.challenge({ 
+            factorId: '00000000-0000-0000-0000-000000000000'
```
Comment (J0) [inline spec/supabase_js_v2.yml:453]: /nit might be slightly clearer if we use a realistic uuid
```suggestion
            factorId: '34e770dd-9ff9-416c-87fa-43b31d7ef225' 
```

File: spec/supabase_js_v2.yml [hunk 2]
```diff
@@ -421,6 +421,80 @@ pages:
...
+  auth.mfa.verify():
+    title: 'mfa.verify()'
+    $ref: '@supabase/gotrue-js.GoTrueMFAApi.verify'
+    examples:
+      - name: Verifies a challenge for a factor
+        isSpotlight: true
+        js: |
+          ```js
+          const { data, error } = await supabase.auth.mfa.verify({ 
+            factorId: '00000000-0000-0000-0000-000000000000',
```
Comment (J0) [inline spec/supabase_js_v2.yml:465]: same as above

File: spec/supabase_js_v2.yml [hunk 3]
```diff
@@ -421,6 +421,80 @@ pages:
...
+          const { data, error } = await supabase.auth.mfa.challenge({ 
+            factorId: '00000000-0000-0000-0000-000000000000' 
+          })
+          ```
+  auth.mfa.verify():
+    title: 'mfa.verify()'
+    $ref: '@supabase/gotrue-js.GoTrueMFAApi.verify'
+    examples:
+      - name: Verifies a challenge for a factor
+        isSpotlight: true
```
Comment (J0) [inline spec/supabase_js_v2.yml:461]:
```suggestion
```

Not sure if we're keeping the `isSpotlight` tag - iirc they're removed in the latest update

cc: @MildTomato for confirmation

File: spec/supabase_js_v2.yml [hunk 4]
```diff
@@ -421,6 +421,80 @@ pages:
...
+    $ref: '@supabase/gotrue-js.GoTrueMFAApi.challengeAndVerify'
+    notes: |
+      - Combines `mfa.challenge()` and `mfa.verify()` in 1 step
+    examples:
+      - name: Creates and verifies a challenge for a factor
+        isSpotlight: true
+        js: |
+          ```js
+          const { data, error } = await supabase.auth.mfa.challengeAndVerify({
+            factorId: '00000000-0000-0000-0000-000000000000',
```
Comment (J0) [inline spec/supabase_js_v2.yml:481]: /nit same thing about uuid
```suggestion
            factorId: '34e770dd-9ff9-416c-87fa-43b31d7ef225',
```

File: spec/supabase_js_v2.yml [hunk 5]
```diff
@@ -421,6 +421,80 @@ pages:
...
+            code: '123456'
+          })
+          ```
+  auth.mfa.getAuthenticatorAssuranceLevel():
+    title: 'mfa.getAuthenticatorAssuranceLevel()'
+    $ref: '@supabase/gotrue-js.GoTrueMFAApi.getAuthenticatorAssuranceLevel'
+    notes:
+      - Authenticator Assurance Level (AAL) is the measure of the strength of an authentication mechanism.
```
Comment (J0) [inline spec/supabase_js_v2.yml:489]: might want to add the optional `refreshSession` param in here if we plan on introducing it-  can do it a  FLUP PR if that helps

File: spec/supabase_js_v2.yml [hunk 6]
```diff
@@ -681,6 +755,31 @@ pages:
...
+  auth.admin.mfa.deleteFactor():
+    title: 'mfa.deleteFactor()'
+    $ref: '@supabase/gotrue-js.GoTrueAdminMFAApi.deleteFactor'
+    examples:
+      - name: Deletes a factor associated to a user
+        isSpotlight: true
+        js: |
+          ```js
+          const { data, error } = await supabase.auth.admin.mfa.deleteFactor({ 
+            id: '00000000-0000-0000-0000-000000000000',
```
Comment (J0) [inline spec/supabase_js_v2.yml:778]:
```suggestion
            id: '34e770dd-9ff9-416c-87fa-43b31d7ef225', 
            userId: '1f30cbf8-61c3-4622-a878-7be8f949d201'
```

File: spec/supabase_js_v2.yml [hunk 7]
```diff
@@ -681,6 +755,31 @@ pages:
...
+    title: 'mfa.deleteFactor()'
+    $ref: '@supabase/gotrue-js.GoTrueAdminMFAApi.deleteFactor'
+    examples:
+      - name: Deletes a factor associated to a user
+        isSpotlight: true
+        js: |
+          ```js
+          const { data, error } = await supabase.auth.admin.mfa.deleteFactor({ 
+            id: '00000000-0000-0000-0000-000000000000', 
+            userId: '11111111-1111-1111-1111-111111111111'
```
Comment (J0) [inline spec/supabase_js_v2.yml:779]: /nit
```suggestion
            userId: '1f30cbf8-61c3-4622-a878-7be8f949d201'
```

**Review by J0 — APPROVED** [review]

**Comment by kangmingtay** [comment]
closing in favour of #10838 - i've addressed the comments in the review in that new PR too

**Pull request closed without merging**